packages/package_info/lib/package_info.dart

```diff
@@ -24,30 +24,29 @@ class PackageInfo {
   /// See [fromPlatform] for the right API to get a [PackageInfo] that's
   /// actually populated with real data.
   PackageInfo({
-    this.appName,
-    this.packageName,
-    this.version,
-    this.buildNumber,
+    required this.appName,
+    required this.packageName,
+    required this.version,
+    required this.buildNumber,
   });
 
-  static PackageInfo _fromPlatform;
+  static PackageInfo? _fromPlatform;
 
   /// Retrieves package information from the platform.
   /// The result is cached.
   static Future<PackageInfo> fromPlatform() async {
     if (_fromPlatform != null) {
-      return _fromPlatform;
+      return _fromPlatform!;
     }
 
-    final Map<String, dynamic> map =
+    final Map<String, dynamic>? map =
         await _kChannel.invokeMapMethod<String, dynamic>('getAll');
-    _fromPlatform = PackageInfo(
-      appName: map["appName"],
+    return _fromPlatform = PackageInfo(
+      appName: map!["appName"],
```

Comment:
why is `map!` required here, but not below?

Reply:
I'm new to null safety, but it looks like you only have to null check the value once within the same method.

Reply:
ah, what about:
```dart
final Map<String, dynamic> map = await _kChannel.invokeMapMethod<String, dynamic>('getAll')!;
```

Reply:
I tried this as well, but that only null checks the `Future` and not the value the `Future` returns. You would have to add parentheses around the `await`. Something like:

```dart
final Map<String, dynamic> map = (await _kChannel.invokeMapMethod<String, dynamic>('getAll'))!;
```

I wasn't really sure if this was cleaner or not.

Reply:
correct. 

```dart
final Map<String, dynamic> map = (await _kChannel.invokeMapMethod<String, dynamic>('getAll'))!;
```
lgtm